src/sound/sampleGenerator.ts

```diff
@@ -0,0 +1,188 @@
+/**
+ * Sample Generator for the new 370-byte chunk-based sound system
+ * 
+ * This module provides the interface and implementations for generating
+ * exactly 370 bytes of 8-bit unsigned audio data at 22.2kHz.
+ * 
+ * The 370-byte chunk size matches the original Mac game's audio buffer
+ * architecture while adapting to modern browser audio APIs.
+ */
+
+/**
+ * Sample rate of the original Mac game audio
+ */
+export const SAMPLE_RATE = 22200; // 22.2kHz
+
+/**
+ * Number of bytes per chunk - must match original Mac buffer size
+ */
+export const CHUNK_SIZE = 370;
+
+/**
+ * Center value for 8-bit unsigned audio (silence)
+ */
+export const CENTER_VALUE = 128; // 0x80
+
+/**
+ * Interface for sample generators
+ * Each generator must produce exactly 370 bytes of 8-bit unsigned audio
+ */
+export type SampleGenerator = {
+  /**
+   * Generate the next chunk of audio samples
+   * @returns Uint8Array of exactly 370 bytes (8-bit unsigned, 0-255 range)
+   */
+  generateChunk(): Uint8Array;
+  
+  /**
+   * Reset the generator to its initial state
+   */
+  reset(): void;
+}
+
```

Comment:
Convert all classes to builder functions that use the type SampleGeneratorBuilder = ()=>SampleGenerator, e.g., const buildSilenceGenerator:SampleGeneratorBuilder = ()=> {...}